File: web/schemas/admin/insights.py
from datetime import date
from typing import List, Optional

from pydantic import BaseModel


class InsightsOverview(BaseModel):
    total_users: int
    users_with_subscription: int
    active_users: int
    paid_users: int
    no_purchase_users: int
    total_revenue: float
    total_payments: int
    avg_check: float
    conversion_to_paid_pct: float
    trial_users: int
    paying_share_pct: float


class TimeseriesPoint(BaseModel):
    date: date
    new_users: int
    new_subscriptions: int
    revenue: float
    payments_count: int


class InsightsTimeseriesResponse(BaseModel):
    period_from: date
    period_to: date
    points: List[TimeseriesPoint]

File: frontend/src/api/admin/insights.ts
import { apiRequest } from '@/api/client'

export interface InsightsOverview {
  total_users: number
  users_with_subscription: number
  active_users: number
  paid_users: number
  no_purchase_users: number
  total_revenue: number
  total_payments: number
  avg_check: number
  conversion_to_paid_pct: number
  trial_users: number
  paying_share_pct: number
}

export interface TimeseriesPoint {
  date: string
  new_users: number
  new_subscriptions: number
  revenue: number
  payments_count: number
}

export interface InsightsTimeseriesResponse {
  period_from: string
  period_to: string
  points: TimeseriesPoint[]
}

export function getInsightsOverview(): Promise<InsightsOverview> {
  return apiRequest<InsightsOverview>('/admin/insights/overview')
}

export function getInsightsTimeseries(days = 90, period_from?: string, period_to?: string): Promise<InsightsTimeseriesResponse> {
  const params = new URLSearchParams()
  if (period_from) params.set('period_from', period_from)
  if (period_to) params.set('period_to', period_to)
  if (!period_from && !period_to) params.set('days', String(days))
  return apiRequest<InsightsTimeseriesResponse>(`/admin/insights/timeseries?${params.toString()}`)
}
